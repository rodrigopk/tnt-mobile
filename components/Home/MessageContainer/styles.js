import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex:1,
    backgroundColor: '#F5F5F5',
    paddingTop:20,
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
  },
  title:{
    fontSize: 32,
  },
  message:{
    fontSize: 21,
  },
});

export default styles;
