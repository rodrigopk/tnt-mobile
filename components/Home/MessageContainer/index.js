import React, { Component } from 'react';
import { View, Text } from 'react-native';

import styles from './styles';

const MessageContainer = ({ title, message }) => (
  <View style={styles.container}>
    <Text style={styles.title}>
      { title }
    </Text>
    <Text style={styles.message}>
      { message }
    </Text>
  </View>
);

export default MessageContainer;
