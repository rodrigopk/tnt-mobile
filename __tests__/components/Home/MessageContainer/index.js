import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';

import MessageContainer from '../../../../components/Home/MessageContainer';

describe('MessageContainer component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <MessageContainer message={'Some message'} title={'Some title'} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
