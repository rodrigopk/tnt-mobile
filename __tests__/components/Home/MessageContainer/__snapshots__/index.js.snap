// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContainer component renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MessageContainer
    message="Some message"
    title="Some title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 32,
            }
          }
        >
          Some title
        </Text>,
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 21,
            }
          }
        >
          Some message
        </Text>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#F5F5F5",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "paddingTop": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Some title",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 32,
          },
        },
        "ref": null,
        "rendered": "Some title",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Some message",
          "ellipsizeMode": "tail",
          "style": Object {
            "fontSize": 21,
          },
        },
        "ref": null,
        "rendered": "Some message",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 32,
              }
            }
          >
            Some title
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 21,
              }
            }
          >
            Some message
          </Text>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#F5F5F5",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingTop": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Some title",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 32,
            },
          },
          "ref": null,
          "rendered": "Some title",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Some message",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 21,
            },
          },
          "ref": null,
          "rendered": "Some message",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
